src/dsl/cb.rs

```diff
@@ -11,18 +11,46 @@ use super::StepTypeHandler;
 pub struct Constraint<F> {
     pub annotation: String,
     pub expr: Expr<F>,
+    pub typing: Typing,
+}
+
+#[derive(Clone)]
+pub enum Typing {
+    Unknown,
+    Boolean,
+    AntiBooly
+}
+
+impl Debug for Typing {
```

Comment:
I think you can derive this, or there is a reason why you want to do it this way?

Reply:
Derived!